source/architecture.rst

```diff
@@ -1,230 +1,245 @@
-communication with each one of the services, an orchestration service to abstract the middleware configuration, an
-authentication service to validate user access credentials and also an user application with graphical interface with
-the purpose of managing the middleware (users, devices and data flows management).
-
-Considering this architecture the basic utilization flow is as follows: The user configures IoT devices through the
-GUI or directly using the REST APIs provided by the API Gateway, he also configures processing flows to deal with the
-data generated by the configured devices. As an example we can generate notifications when the data of a device has one
-of its values reaching a threshold or we can add an entry to a database when a device enters or leaves an specific
-geographic location. This user operations on the API result in configurations across the internal services of the
-middleware (Broker, CEP and iot-agent), being partially abstracted by the orchestration service.
-
-The user contexts are isolated and there is no data sharing, the access credentials are validated by the authorization
-service for each and every operation (API Request). Once the devices and the flows are configured, the iot-agent is
-capable of mapping the data received from devices, encapsulated on MQTT for example, and send then to the context broker
-for internal distribution, reaching, for instance, the history service so it can persist the data on a database and the
-CEP for processing it based on rules. If certain conditions are matched when rules are being processed, a new event is
-generated and sent to the broker service to be redistributed to the interested services.
-
-This architecture made possible the validation of ideas and limitations and possible improvements were identified,
-converging to a reviewed architectural proposition as described on :numref:`new_arch`. This new proposal is under
-development and should become part of the solution.
+dojot was designed to make fast solution prototyping possible, providing a
+platform that's easy to use, scalable and robust. Its internal architecture
+makes use of many well-known open-source components with others designed and
+implemented by dojot team. This architecture is described on
+:numref:`new_arch`.
 
 .. _new_arch:
 .. figure:: images/new_architecture.png
     :width: 100%
     :align: center
     :alt: Revised *dojot* Architecture
 
-    Revised Architecture
-
-More detailed and updated information can be found on the `dojot Github repository <https://github.com/dojot>`_.
-
-Each one of the components that are part of the architecture are briefly described on the sub-sections below.
-
-Kafka + Subscription Manager + NGSI
+    Current Architecture
+
+Considering this architecture the basic utilization flow is as follows: The
+user configures IoT devices through the GUI or directly using the REST APIs
+provided by the API Gateway, he also configures processing flows to deal with
```

Comment:
The user configures devices using the REST APIs provided by the API Gateway, that for ease of use are graphically presented by the GUI
---
I think that the REST APIs should be highlighted as the main access point because the GUI is just an exhibition/example app and any actual app that is built by a third party will make use of the APIs directly.

Also, the text says: "considering the architecture" where the GUI is not shown as a component

Reply:
Ok. I updated the architecture image including a GUI block. Regarding the text, how about: 
```
Using dojot is as follows: a user configures IoT devices through the GUI or
directly using the REST APIs provided by the API Gateway. Data processing flows
might be also configured - these entities can perform a variety of actions,
such as generate notifications when a particular device attribute reaches a
certain threshold or save all data generated by a device onto an external
database. As devices start sending their readings to dojot, the user might want
to receive these readings via notifications generated by subscriptions,
consolidate all data into virtual devices, gather all data from historical
database, and so on. These features can be used through REST APIs - these are
the basic building blocks that any application based on dojot should use. dojot
GUI provides an easy way to perform management operations for all entities
related to the platform (users, devices, templates and flows) and can also be
used to check if everything is working fine.
```